src/slice-assign.c

```diff
@@ -32,10 +45,8 @@ r_obj* vec_assign_opts(r_obj* x,
   value = KEEP(vec_check_recycle(value, vec_size(index), opts.value_arg, opts.call));
 
   r_obj* proxy = KEEP(vec_proxy(x));
-  const enum vctrs_owned owned = vec_owned(proxy);
-  proxy = KEEP(vec_proxy_assign_opts(proxy, index, value, owned, &opts));
-
-  r_obj* out = vec_restore(proxy, x, owned);
+  proxy = KEEP(vec_proxy_assign_opts(proxy, index, value, &opts));
+  r_obj* out = vec_restore(proxy, x);
```

Comment:
This is where my saga started:
- `vec_proxy()` on a POSIXlt returns a "fresh" data frame, but note that the underlying columns aren't "fresh"
- That data frame looks "owned" by us though, so `owned = VCTRS_OWNED_true`
- That propagates through all the column level assignment in `df_assign()`, so we overwrite the underlying columns in place 😱